konfera/utils.py

```diff
@@ -41,3 +49,48 @@ def update_event_context(event, context, show_sponsors=True):
 
 def currency_round_up(money):
     return money.quantize(Decimal('1.00'), rounding=ROUND_UP)
+
+
+def validate_email_template(raw_template, input):
+    required_keys = set(re.findall('{(.+?)}', raw_template))
+    if required_keys.issubset(set(input.keys())):
+        return raw_template.format(**input)
+    return
+
+
+class EmailTemplateError(Exception):
+    pass
+
+
+def send_email(addresses, subject, template, input=None, verbose=0):
+    input = input or {}
+    text_template = getattr(template, 'text_template', '')
+    html_template = getattr(template, 'html_template', '')
+    text_content = validate_email_template(text_template, input)
+    html_content = validate_email_template(html_template, input)
+
+    if not text_content:
+        logger.critical('Invalid text template (required) for the input {}.'.format(text_content))
+        # raise EmailTemplateError("Email template is not valid for the input.")
+    if not html_content:
+        logger.warning('Invalid html template (not required) for the input {}.'.format(html_content))
+
+    to = addresses.get('to', [])
+    cc = addresses.get('cc', [])
+    bcc = addresses.get('bcc', EMAIL_NOTIFY_BCC)
+
+    # if text template does not match - do we want to raid the exception or to send unformatted template?
```

Comment:
I think exception should be here, because:

If there is an error in template first who encounters it will raise exception, and admin should amend accordingly (he has a chance to fix after first encounter). If he does not exceptions will always be raised and user does not recaive email, on the other hand if we always send email with wrong text, user will not receive any useful info (to me it is almost the same as not receive email at all), and admin will not see this in logs, monitoring, anywhere so he doesnt have the chance to fix. And in this case fix is re-wording the template in DB.